api/rolesApi.ts

```diff
@@ -0,0 +1,17 @@
+import { ApiPromise } from '../types/apiResponse';
+import { RoleData, RolePostData } from '../types/role';
+import api from './api';
+
+class RolesApi {
+  private getRolesUrl() {
+    return 'roles';
+  }
+
+  public createRole(role: RolePostData): ApiPromise<RoleData> {
+    return api.post(this.getRolesUrl(), role).then((res) => res.data);
+  }
+}
+
+const rolesApi = new RolesApi();
```

Comment:
Btw Hong Yao, is there a reason we want to use classes? I feel the functional way has less boiler plate and easier to write.

```suggestion
const ROLES_URL = 'roles';

export function createRole(role: RolePostData): ApiPromise<RoleData> {
   return api.post(ROLES_URL, role).then((res) => res.data);
}

// export function getRoles(companyId: number, searchQuery?: string): ApiPromise<RoleData> {
//   return api.get(...);
// }
```

```
rolesApi.createRole(...)
```

Reply:
I feel that having an additional namespace makes it clear that this is an API call. There may be more similarly named functions down the road such as Redux actions and functions within your components. What do you think about this?

Reply:
Makes sense. Can we keep the roles url as a const instead of a function then?